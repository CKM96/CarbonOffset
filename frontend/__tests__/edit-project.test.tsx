import { fireEvent, render } from '@testing-library/react';
import EditProject from '../pages/edit-project/[id]';
import fetchMock from 'jest-fetch-mock';
import { act } from 'react-dom/test-utils';

jest.mock('next/router', () => require('next-router-mock'));

afterEach(jest.clearAllMocks);

describe('CreateProject', () => {
  it('matches snapshot', async () => {
    const { container, findByText } = render(<EditProject />);

    await findByText('Edit Offsetting Project');

    expect(container).toMatchSnapshot();
  });

  it('creates a new project', async () => {
    fetchMock.mockOnce('Created', { status: 201 });

    const { getByLabelText, getByText } = render(<EditProject />);
    
    fireEvent.change(getByLabelText('Name'), {
      target: { value: 'Project name' },
    });
    fireEvent.change(getByLabelText('Description (optional)'), {
      target: { value: 'Project description' },
    });

    await act(async () => {fireEvent.click(getByText('Submit'))});

    expect(fetchMock.mock.calls.length).toEqual(1);
    expect(fetchMock.mock.calls[0][0]).toEqual(
      'http://localhost:3001/projects',
    );
    expect(fetchMock.mock.calls[0][1]?.method).toEqual('put');
    expect(fetchMock.mock.calls[0][1]?.body).toEqual(
      JSON.stringify({
        name: 'Project name',
        description: 'Project description',
      }),
    );
  });
});
