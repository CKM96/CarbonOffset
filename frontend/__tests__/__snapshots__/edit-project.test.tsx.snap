// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateProject matches snapshot 1`] = `
<div>
  <h1>
    Edit Offsetting Project
  </h1>
  <form
    class="sc-bczRLJ goEvZo"
  >
    <label
      class="sc-gsnTZi fioPzC"
    >
      Name
      <input
        required=""
        type="text"
        value=""
      />
    </label>
    <label
      class="sc-gsnTZi fioPzC"
    >
      Description (optional)
      <input
        type="text"
        value=""
      />
    </label>
    <input
      class="sc-dkzDqf ebizIa"
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;
